src/lib.rs

```diff
@@ -425,7 +438,17 @@ pub trait PolynomialCommitment<G: AffineCurve>: Sized {
         Ok((combined_polynomial, combined_randomness, has_hiding))
     }
     
-    /// Build multi poly multi point opening combination
+    /// Build the multi-point multi-poly opening combination. 
+    /// This is the Boneh et al. method of proving a multi-point multi-poly claim
+    ///     `p_i(x_i) = y_i`, `i=1,...,m`, 
+    /// by a polynomial identity over the query set {x_1,...,x_m}
+    ///      Sum_i  lambda^i z_i(X) * (p_i(X) - y_i)  = h(X) * z(X),
+    /// where `z_i(X) = Prod_{x_j!=x_i} (x - x_j)` and `z(X)` is the vanishing 
+    /// polynomial of the query set. The identity is assured by
+    /// providing h(X) and showing that for a random x outside the query set, 
+    /// the "mult-point multi-poly opening combination"
+    ///       LC_x(p_i(X),h(X)) = Sum_i  lambda^i z_i(x)/z(x) * p_i(X)  - h(X)
+    /// opens to the expected value `Sum_i lambda^i z_i(x)/z(x)* y_i`.
     fn multi_point_multi_poly_open_combine<'a>(
```

Comment:
Could the body of this function be compressed?
There are two cycles for on the same set:_
"for (label, (_point_label, point)) in query_set.iter()"
In both this cycles the x_polynomial is redefined in the same way.